### PR TITLE
HotFix Dev-1937 Show All Fed Accounts in TreeMap :)

### PR DESCRIPTION
**Pull request by jonhill13**
**High level description:**

TreeMap now displays all positive Federal Accounts.

**Technical details:**

* add for await to code base

* give fed account tree own no results, loading and error message

* give fed account table own no results, loading and error message

* new method to page through and get all fed accounts

* new method to parse all fed accounts

**JIRA Ticket:**
[DEV-1937](https://federal-spending-transparency.atlassian.net/browse/DEV-1937)

**Mockup:**
https://bahdigital.invisionapp.com/share/69IA8EPGPCM#/screens/296001079

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] Design review @AGuyNamedMarco 
- [x] [API #1803](https://github.com/fedspendingtransparency/usaspending-api/pull/1803) merged
- [x] Verified cross-browser compatibility

### CODE REVIEW

File: src/js/containers/awardV2/idv/AwardMetaDataContainer.jsx [hunk 1]
```diff
@@ -78,6 +86,47 @@ export class AwardMetaDataContainer extends React.Component {
...
     }
 
+    async getAllFederalAccounts() {
+        if (this.allFederalAccountsRequest) {
+            this.allFederalAccountsRequest.cancel();
+        }
+        this.setState({ inFlightTreemap: true, errorTreemap: false });
+        async function* paginationFunction(awardId) {
+            let hasNext = true;
+            const limit = 10;
```
Comment (tony-sappe) [inline src/js/containers/awardV2/idv/AwardMetaDataContainer.jsx]: Good idea to bump this up to 100

**Comment by MaxwellKendall** [comment]
#### A couple styling issues I noticed when no Chart is available:
- List view doesn't look right to me when there are no results:
1. Federal Accounts header is not present
2. List view column headers are underneath the Chart Not Available place holder
image

- Tree map view 
1. Lots of space underneath and above the chart
image

Overall, really like the use of the generator function to handle the fetching of all the accounts. Would like to hear you talk through that as well! Let me know if you need an extra pair of eyes to look at some of this!

File: src/js/containers/awardV2/idv/AwardMetaDataContainer.jsx (hunk 1 above)
Comment (jonhill13) [reply src/js/containers/awardV2/idv/AwardMetaDataContainer.jsx]: 🥇

**Comment by jonhill13** [comment]
@MaxwellKendall thank you for taking a look, definitely should have double checked my rendering situations.  We can talk about generators anytime.
*List View*
1. Federal Accounts header is not present
- this was not in the mock, unless that has changed, it is only in the TreeMap mock
2. List view column headers are underneath the Chart Not Available place holder
- the new changes should handle that
*TreeMap*
1. Lots of space underneath and above the chart
- the new changes should handle that, looks like something was holding that space

File: src/js/components/awardv2/idv/federalAccounts/FederalAccountsTable.jsx [hunk 2]
```diff
@@ -94,8 +99,17 @@ export default class FederalAccountsTable extends React.Component {
     }
 
     render() {
+        const { inFlight, error, federalAccounts } = this.props;
         return (
             <div className="federal-accounts-table-holder">
+                <div className="results-table-message-container">
+                    {inFlight && <ResultsTableLoadingMessage />}
+                    {(error && !inFlight) && <ResultsTableErrorMessage />}
```
[the remark this answers is not in the document]
Comment (jonhill13) [reply src/js/components/awardv2/idv/federalAccounts/FederalAccountsTable.jsx]: Thanks, I changed all of it with the new code

File: src/js/containers/awardV2/idv/AwardMetaDataContainer.jsx [hunk 3]
```diff
@@ -78,6 +86,47 @@ export class AwardMetaDataContainer extends React.Component {
...
+                yield response.data;
+            }
+        };
+        async function handleResponse(awardId) {
+            const iterator = paginationFunction(awardId);
+            const allFederalAccounts = [];
+            let totalAccounts = 0;
+            for await (const federalAccounts of iterator) {
+                allFederalAccounts.push(...federalAccounts.results);
+                totalAccounts = federalAccounts.page_metadata.count;
```
[the remark this answers is not in the document]
Comment (jonhill13) [reply src/js/containers/awardV2/idv/AwardMetaDataContainer.jsx]: no, it does not need to exist at all :)

File: babel.config.js [hunk 4]
```diff
@@ -20,6 +20,7 @@ const plugins = [
     ],
     "@babel/plugin-transform-runtime",
     "@babel/plugin-proposal-object-rest-spread",
+    "@babel/plugin-proposal-async-generator-functions",
     "react-hot-loader/babel"
```
[the remark this answers is not in the document]
Comment (jonhill13) [reply babel.config.js]: I guess not, seems to be working fine without it

File: package.json [hunk 5]
```diff
@@ -39,7 +39,7 @@
     "git-revision-webpack-plugin": "^3.0.3",
     "html-webpack-plugin": "^3.2.0",
     "identity-obj-proxy": "^3.0.0",
-    "jest": "^24.6.0",
+    "jest": "^24.8.0",
     "jest-cli": "^24.6.0",
```
Comment (MaxwellKendall) [inline package.json]: Was this intentional? Don't think it really matters but not sure what the  protocol is on updating major packages in our code base is... (cc: @ebdabbs) If we are updating this, I would recommend keeping the `jest-cli` package on the same version!

File: src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx [hunk 6]
```diff
@@ -30,7 +30,7 @@ const FederalAccountsSummary = ({
                     </span>
                 </div>
                 <div className="award-funding-summary__data">
-                    <span>Total Count Of Funding Agencies</span>
+                    <span>Total Count of Funding Agencies</span>
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx]: 👌

Comment (MaxwellKendall) [reply src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx]: I think there is actually a ticket for this as well?

File: package.json (hunk 5 above)
Comment (jonhill13) [reply package.json]: Prolly left over from getting the tests to work

File: src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx (hunk 6 above)
Comment (jonhill13) [reply src/js/components/awardv2/idv/federalAccounts/FederalAccountsSummary.jsx]: not that I am aware of. I also checked with Melissa.

**Pull request merged**